src/huggingface_hub/inference/_client.py

```diff
@@ -813,16 +800,28 @@ def _resolve_url(self, model: Optional[str] = None, task: Optional[str] = None)
 
 
 def _get_recommended_model(task: str) -> str:
-    # TODO: load from a config file? (from the Hub?) Would make sense to make updates easier.
-    if task in RECOMMENDED_MODELS:
-        model = RECOMMENDED_MODELS[task]
-        logger.info(
-            f"Defaulting to recommended model {model} for task {task}. It is recommended to explicitly pass"
-            f" `model='{model}'` as argument as we do not guarantee that the recommended model will stay the same over"
-            " time."
+    model = _fetch_recommended_models().get(task)
+    if model is None:
+        raise ValueError(
+            f"Task {task} has no recommended task. Please specify a model explicitly. Visit"
+            " https://huggingface.co/tasks for more info."
         )
-        return model
-    raise NotImplementedError()
+    logger.info(
+        f"Using recommended model {model} for task {task}. Note that it is encouraged to explicitly set"
+        f" `model='{model}'` as the recommended models list might get updated without prior notice."
+    )
+    return model
+
+
+def _fetch_recommended_models() -> Dict[str, Optional[str]]:
+    global _RECOMMENDED_MODELS
+    if _RECOMMENDED_MODELS is None:
+        response = get_session().get("https://huggingface.co/api/tasks", headers=build_hf_headers())
+        hf_raise_for_status(response)
+        _RECOMMENDED_MODELS = {
+            task: _first_or_none(details["widgetModels"]) for task, details in response.json().items()
```

Comment:
Alternatively (no strong suggestion)

```suggestion
            task: next(iter(details["widgetModels"])) for task, details in response.json().items()
```

Reply:
Yeah, not exactly because `details["widgetModels"]` can alos be None :confused: